docs/build/use-cases/trustless-agent.mdx

```diff
@@ -0,0 +1,308 @@
+
+You will need:
+- Docker (or Podman)
+- [Oasis CLI]
+- Node.js 18+ (for Eliza and helper scripts)
+- a Sapphire Testnet wallet funded with TEST
+
+Optional depending on your LLM choice:
+- OpenAI API key or a reachable Ollama instance
+
+[Oasis CLI]: https://github.com/oasisprotocol/cli/blob/master/docs/README.md
+
+## ROFL overview
+
+## Architecture Overview
+
+- ROFL (Runtime Off-chain Logic)
+  - Enclave execution, sealed secrets, app identity, attestation
+  - In-enclave key manager via `appd` UNIX socket
+- Sapphire (Verification Layer)
+  - Confidential EVM chain verifying ROFL identities and policies
+  - Runs confidential contracts that can verify ROFL origin
+
+## Part 1: ROFL
+
+We package the agent as a container and deploy it to a TEE via ROFL. ROFL
+handles attestation, sealed secrets, and app identity bound to your on-chain
+registration.
+
+### Initialize and create the ROFL app
+
+```shell
+oasis rofl init rofl-eliza
+cd rofl-eliza
+oasis rofl create --network testnet
+```
+
+After creation, note your ROFL app ID. You can monitor deployments with:
+`oasis rofl machine show` and `oasis rofl machine logs`, or use the Oasis
+Explorer on Sapphire Testnet.
+
+## Eliza example (local run)
+
+This section outlines a minimal Eliza setup that you can later deploy in ROFL.
+
+1. Set up a simple Eliza project (or use your existing one).
+2. Configure environment variables for your LLM backend.
+3. Run locally to verify the agent works.
+4. Containerize and deploy to ROFL (see sections below).
+
+#### 1) Initialize project
+
+```shell
+mkdir eliza-app && cd eliza-app
+npm init -y
+# install your Eliza framework deps here
+# npm install @your-eliza/pkg ...
+```
+
+Initialize a project using the ElizaOS CLI and prepare it for ROFL.
+
+<details>
+  <summary>ElizaOS CLI quickstart (recommended)</summary>
+
+```shell
+# 1) Install bun and ElizaOS CLI
+bun --version || curl -fsSL https://bun.sh/install | bash
+bun install -g @elizaos/cli
+
+# 2) Scaffold a project
+elizaos create eliza-agent
+cd eliza-agent
+
+# 3) Configure model credentials locally
+elizaos env edit-local
+# Add e.g.:
+#   OPENAI_API_KEY=sk-...
+# or for local models:
+#   OLLAMA_HOST=http://host.docker.internal:11434
+
+# 4) Start the agent locally
+elizaos start
+```
+</details>
+
+#### Containerize and deploy to ROFL
+
+Proceed to “Eliza agent container (inside ROFL)” below to package the agent
+for enclave execution and deploy it using the Oasis CLI.
+
+## Part 3: Eliza agent container (inside ROFL)
+
+Create a `docker` folder with a minimal runner that:
+1) starts an ElizaOS agent via the CLI and
+2) reads model configuration from environment variables.
+
+<details>
+  <summary>docker/entry.sh</summary>
+
+```shell
+#!/bin/sh
+set -e
+
+# Expected env in ROFL:
+#   OPENAI_API_KEY (or OLLAMA_HOST) for the model provider
+# Provide any additional vars your agent needs here.
+exec elizaos start
+```
+</details>
+
+<details>
+  <summary>docker/appd-helper.js</summary>
```

Comment:
We can ditch this file by using the new @oasisprotocol/rofl-client right https://github.com/oasisprotocol/oasis-sdk/tree/main/rofl-client/ts?